exchanges/coinbasepro/coinbasepro_wrapper.go

```diff
@@ -632,19 +633,60 @@ func (c *CoinbasePro) AuthenticateWebsocket() error {
 	return common.ErrFunctionNotSupported
 }
 
-// GetHistoricCandles Allows to retrieve an amount of candles back in time starting from now up to rangesize * granularity, where granularity is the trade period covered by each candle
-func (c *CoinbasePro) GetHistoricCandles(p currency.Pair, rangesize, granularity int64) ([]exchange.Candle, error) {
-	end := time.Now().UTC()
-	b := granularity * rangesize
-	start := time.Now().UTC().Add(-time.Second * time.Duration(b))
-	history, err := c.GetHistoricRates(p.String(), start.Format(time.RFC3339), end.Format(time.RFC3339), granularity)
+// checkInterval checks allowable interval and if it exceeds 300 possible return
+// entries will throw an error
+func checkInterval(i time.Duration, start, end time.Time) (int64, error) {
+	var counter int
+	for newstart := start.Truncate(i); newstart.Before(end.Truncate(i)); newstart = newstart.Add(i) {
+		counter++
+	}
+
+	if counter > 300 {
+		return 0, errors.New("potential candles exceed 300 please reduce time between start and end")
```

Comment:
I don't think we should have this. As per the api docs: 

> The maximum number of data points for a single request is 300 candles. If your selection of start/end time and granularity will result in more than 300 data points, your request will be rejected. If you wish to retrieve fine granularity data over a larger time range, you will need to make multiple requests with new start/end ranges.

It will get rejected for us. no need to internalise their current API policy because it is prone to change. If they ever increase this number, we won't be able to know.


... Unless the error returned doesn't specify the reason why, in which case, keep this.

Reply:
`{"message":"granularity too small for the requested time range. Count of aggregations requested exceeds 300"}`

Yeah I will RM 👍